second.ino

```diff
@@ -60,15 +60,15 @@ float dt;
 float milliOld_mpu;
 float milliNew_mpu;
 float dt_mpu;
-int error = 0;
+int error = 0; // ?? Why there are so many global variables here while we can use them in functions only 
 float previousError = 0;
 float ErrorChange;
 float Slope;
 float Area = 0;
 int pid;
-int counter = 0;
-char arr[5] = {'f', 'f', 'r', 'r', 'r'};
-int d[5];
+int counter = 0;// ?? What is the usage of this counter
+//char arr[5] = {'f', 'f', 'r', 'r', 'r'}; // ?? this array shouldn't be here it should be in void loop
```

Comment:
This was left a global variable for the following reasons:

1. Pointers are Hard to understand to some people :(
2. If an array is passed to a function, it will be passed as a pointer to its first location and `sizeof` operator will not work as expected, so it was best to  leave it as a global variable.